src/Octoshift/HttpDownloadServiceFactory.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Net.Http;
+using OctoshiftCLI.Contracts;
+
+namespace OctoshiftCLI
+{
+    public sealed class HttpDownloadServiceFactory
+    {
+        private readonly OctoLogger _log;
+        private readonly IHttpClientFactory _clientFactory;
+        private readonly FileSystemProvider _fileSystemProvider;
+        private readonly IVersionProvider _versionProvider;
+
+        public HttpDownloadServiceFactory(OctoLogger log, IHttpClientFactory clientFactory, FileSystemProvider fileSystemProvider, IVersionProvider versionProvider)
+        {
+            _log = log;
+            _clientFactory = clientFactory;
+            _fileSystemProvider = fileSystemProvider;
+            _versionProvider = versionProvider;
```

Comment:
this can be removed now since we're not injecting it anymore.

Reply:
The `versionProvider`? Don't we still need to inject it here so we can inject it into the service for the httpclient default request headers [(HttpDownloadService, line 27)](https://github.com/github/gh-gei/blob/4067972d91d223245bd8ae96b7159b46199173f2/src/Octoshift/HttpDownloadService.cs#LL27C35-L27C35)? Or am I misunderstanding?

Reply:
Oh yeah you're right, I though it was only used for setting the user agent header.